test/basic-querying.test.js

```diff
@@ -620,6 +620,165 @@ describe('basic-querying', function () {
   });
 });
 
+describe('updateOrCreate', function() {
+  var db = getSchema();
+  var Todo;
+
+  before(function setUp(done) {
+    Todo = db.define('Todo', {
+      content: String
+    });
+    db.automigrate('Todo', done);
+  });
+  afterEach(function tearDown(done) {
+    db.automigrate(done);
+  });
+
+  context('single model queries', function() {
+    it('creates a model when one does not exist', function(done) {
+      Todo.updateOrCreate({content: 'a'}, function(err, data) {
+        if (err) return done(err);
+
+        should.exist(data);
+        should.exist(data.content);
+        data.content.should.equal('a');
+
+        done();
+      });
+    });
+
+    it('updates a model if it exists', function(done) {
+      Todo.create({content: 'a'}, function(err, todo) {
+        Todo.updateOrCreate({id: 1, content: 'b'}, function(err, data) {
+          if (err) return done(err);
+
+          should.exist(data);
+          should.exist(data.id)
+          data.id.should.equal( 1);
+          should.exist(data.content);
+          data.content.should.equal('b');
+
+          done();
+        });
+      });
+    });
+
+    //TODO - Error cases
+
+  });
+
+  //TODO batch model queries
+  context('model queries with array input', function() {
+    // Array with single model
+    it('creates a single model if it does not exist - single object input without id', function(done) {
+      Todo.updateOrCreate([{content: 'a'}], function(err, data) {
+        if (err) return done(err);
+
+        should.exist(data);
```

Comment:
Don't need this assertion as the next assertion will check if it exists anyways.